### PR TITLE
scons 4.0.1

### PR DESCRIPTION
**Pull request by SeekingMeaning**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/scons.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 class Scons < Formula
-  include Language::Python::Shebang
+  include Language::Python::Virtualenv
 
   desc "Substitute for classic 'make' tool with autoconf/automake functionality"
   homepage "https://www.scons.org/"
-  url "https://downloads.sourceforge.net/project/scons/scons/3.1.2/scons-3.1.2.tar.gz"
-  sha256 "7801f3f62f654528e272df780be10c0e9337e897650b62ddcee9f39fde13f8fb"
-  revision 2
+  url "https://files.pythonhosted.org/packages/source/S/SCons/SCons-4.0.1.tar.gz"
```
Comment (SMillerDev) [inline Formula/scons.rb:6]: Normally we use a different python resource URL. Could you change it to that format?

Comment (dawidd6) [reply Formula/scons.rb:6]: https://files.pythonhosted.org/packages/ae/a4/2eb8d05b0ac9e168e8ff0681624c123a123c743487e528757c68ea995d20/SCons-4.0.1.tar.gz

Comment (SeekingMeaning) [reply Formula/scons.rb:6]: Why not this format? (Sorry if I’m missing something)

**Comment by SeekingMeaning** [comment]
I should have mentioned in #58130 that I changed the URL format in that PR as well, sorry

File: Formula/scons.rb (hunk 1 above)
Comment (claui) [reply Formula/scons.rb:6]: > Why not this format? (Sorry if I’m missing something)

Off the top of my head:

- Doing things consistently makes it not only more pleasant on the eyes but also easier to debug visually and spot mistakes.
- `files.pythonhosted.org` has an excellent track record in terms of availability so it makes sense to stick with it wherever we can.
- In contrast to a corporation, a non-profit foundation may be less prone to introducing sudden arbitrary download restrictions or similar user-hostile business decisions. One real-world example of this is FOSShub, who in 2016 (?) decided to make their download URLs require single-use tokens in order to deter automated downloads and make more money from ads.
- Reducing unnecessary variance makes it easier to introduce automated audits and to make bulk changes.

Comment (SeekingMeaning) [reply Formula/scons.rb:6]: > * Doing things consistently makes it not only more pleasant on the eyes but also easier to debug visually and spot mistakes.

True, although I think this format is easier to read as the only part of the URL that needs to change when updating is the version, for example:

```patch
-  url "https://files.pythonhosted.org/packages/source/S/SCons/SCons-4.0.1.tar.gz"
+  url "https://files.pythonhosted.org/packages/source/S/SCons/SCons-4.0.2.tar.gz"
```

If necessary, I could comb through all formulae that are hosted on `files.pythonhosted.org` and update them to this new format

> * `files.pythonhosted.org` has an excellent track record in terms of availability so it makes sense to stick with it wherever we can.

But this already is `files.pythonhosted.org`...?

> * In contrast to a corporation, a non-profit foundation may be less prone to introducing sudden arbitrary download restrictions or similar user-hostile business decisions. One real-world example of this is FOSShub, who in 2016 (?) decided to make their download URLs require single-use tokens in order to deter automated downloads and make more money from ads.

Yes, I agree

> * Reducing unnecessary variance makes it easier to introduce automated audits and to make bulk changes.

I agree with this as well, hence I propose changing urls such as:

```
https://files.pythonhosted.org/packages/ae/a4/2eb8d05b0ac9e168e8ff0681624c123a123c743487e528757c68ea995d20/SCons-4.0.1.tar.gz
```

to this format:

```
https://files.pythonhosted.org/packages/source/S/SCons/SCons-4.0.1.tar.gz
```

Comment (claui) [reply Formula/scons.rb:6]: > True, although I think this format is easier to read as the only part of the URL that needs to change when updating is the version, for example:

[…]

> But this already is `files.pythonhosted.org`...?

Apologies. I mistook the quoted diff for a suggestion and thought `downloads.sourceforge.net` was what the current tip of the branch said. Please disregard my comment.

**Comment by SeekingMeaning** [comment]
Changed URL to https://files.pythonhosted.org/packages/ae/a4/2eb8d05b0ac9e168e8ff0681624c123a123c743487e528757c68ea995d20/SCons-4.0.1.tar.gz

This can be updated later when homebrew/brew#8029 is merged

**Review by dawidd6 — APPROVED** [review]

**Comment by claui** [comment]
Thanks @SeekingMeaning! 🍻

**Pull request closed without merging**